infra/modules/azure_role_assignments/modules/apim/locals.tf

```diff
@@ -1,9 +1,14 @@
 locals {
-  apims = distinct([for assignment in var.apim : { name = assignment.name, resource_group_name = assignment.resource_group_name }])
-
+  norm_apims = [for apim in var.apim : {
+    name                = provider::azurerm::parse_resource_id(apim.id)["resource_name"]
```

Comment:
Why do you still need `name` and `resource_group_name`?

Reply:
If they are only internal we can keep them I think. But I wouldn't use them for new submodules